crates/openid4vci/src/credential_issuer/mod.rs

```diff
@@ -502,26 +508,20 @@ impl CredentialIssuer {
             }
         }
 
-        if let Some(credential_offer) = credential_offer {
-            credential_offer.validate()?;
-        } else {
-            return Err(CredentialIssuerError::CredentialOfferMustBeSupplied);
-        };
-
-        if let Some(authorization_server_metadata) = authorization_server_metadata {
-            authorization_server_metadata.validate()?;
-            return Err(CredentialIssuerError::AuthorizationServerMetadataNotSupported);
-        };
-
         if let Some(CredentialRequestProof { jwt, .. }) = &credential_request.proof {
             let jwt = ProofJwt::from_str(jwt)?;
             jwt.validate()?;
 
             let expected_c_nonce = evaluate_credential_request_options
+                .clone()
                 .and_then(|o| o.c_nonce)
                 .map(|c| c.expected_c_nonce);
             jwt.check_nonce(expected_c_nonce)?;
 
+            let expected_issuer = evaluate_credential_request_options.and_then(|o| o.client_id);
+            jwt.check_iss(expected_issuer)?;
```

Comment:
I think the client_id is optional? E.g. in case of pre-authorized flow

> For the Pre-Authorized Code Grant Type, authentication of the client is OPTIONAL, as described in Section 3.2.1 of OAuth 2.0 [[RFC6749](https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#RFC6749)] and consequently, the "client_id" is only needed when a form of Client Authentication that relies on the parameter is used.

Reply:
It seems consistent with what Sphereon has?